userspace/libscap/scap_savefile.c

```diff
@@ -2768,7 +2768,9 @@ int32_t scap_next_offline(scap_t *handle, OUT scap_evt **pevent, OUT uint16_t *p
 			return SCAP_FAILURE;
 		}
 
-		readsize = gzread(f, handle->m_file_evt_buf, readlen);
+		// read a ppm_scap_evt_hdr struct leaving some space at the beginning of the buffer
+		// so that we can cast the whole buffer as ppm_evt_hdr
+		readsize = gzread(f, handle->m_file_evt_buf + sizeof(struct ppm_evt_hdr) - sizeof(struct ppm_scap_evt_hdr), readlen);

```

Comment:
If you really leave 4 bytes at the beginning because the sentinel is enabled, shouldn't you modify also, e.g., line 2779? Otherwise aren't you reading garbage instead of the CPUID? If I'm right, same for the other usages of `handle->m_file_evt_buf`.